/*
 * This file was automatically generated by EvoSuite
 * Mon Apr 10 20:11:40 GMT 2017
 */

package ru.innopolis.lips.memvit.utils;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.TooManyResourcesException;
import org.evosuite.runtime.testdata.EvoSuiteFile;
import org.evosuite.runtime.testdata.FileSystemHandling;
import org.junit.runner.RunWith;
import ru.innopolis.lips.memvit.utils.FileWriter2;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class FileWriter2_ESTest extends FileWriter2_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2();
      fileWriter2_0.writeJson("{H");
      FileSystemHandling.setPermissions((EvoSuiteFile) null, true, true, true);
      fileWriter2_0.writeLog(1, "B69$nJ,");
      fileWriter2_0.writeLog(2720, "1jBm/`XJG0");
      fileWriter2_0.writeLog(2720, "p][va|WRke85US~~\"o");
      fileWriter2_0.writeLog(2961, "{H");
      fileWriter2_0.writeLog(1298, "bky$k");
      // Undeclared exception!
      try { 
        fileWriter2_0.writeLog(2868, "B69$nJ,");
        fail("Expecting exception: TooManyResourcesException");
      
      } catch(TooManyResourcesException e) {
         //
         // Loop has been executed more times than the allowed 10000
         //
         assertThrownBy("org.evosuite.runtime.LoopCounter", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2("");
      fileWriter2_0.writeLog("");
      fileWriter2_0.writeLog((-495), "");
      fileWriter2_0.writeLog((-495), "");
      fileWriter2_0.writeLog("");
      fileWriter2_0.writeLog("");
      fileWriter2_0.writeJson("");
      fileWriter2_0.writeJson("");
      fileWriter2_0.writeLog("");
      fileWriter2_0.writeLog((-495), "");
      FileWriter2 fileWriter2_1 = new FileWriter2();
      FileWriter2 fileWriter2_2 = new FileWriter2("");
      fileWriter2_0.writeLog("");
      fileWriter2_0.writeLog((-495), "");
      FileWriter2 fileWriter2_3 = new FileWriter2();
      fileWriter2_3.writeLog("");
      fileWriter2_3.writeLog(0, "");
      FileWriter2 fileWriter2_4 = new FileWriter2((String) null);
      // Undeclared exception!
      try { 
        fileWriter2_4.writeLog((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("java.io.File", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2("$p]6A&?L+dWO6uPkPJ");
      fileWriter2_0.writeLog(1, "sNoAL");
      fileWriter2_0.writeLog("sNoAL");
      fileWriter2_0.writeJson("XQQ2VsY+&^!@Zkg;@");
      fileWriter2_0.writeLog((-1), "$p]6A&?L+dWO6uPkPJ");
      fileWriter2_0.writeJson("sNoAL");
      // Undeclared exception!
      try { 
        fileWriter2_0.writeJson((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2("");
      fileWriter2_0.writeJson("*fQ'yN,");
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2();
      fileWriter2_0.writeLog((-1011), "");
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2();
      FileWriter2 fileWriter2_1 = new FileWriter2((String) null);
      FileWriter2 fileWriter2_2 = new FileWriter2((String) null);
      // Undeclared exception!
      try { 
        fileWriter2_2.writeLog(300, "dd.MM.yyyy_HH-mm-ss-SSS");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("java.io.File", e);
      }
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      FileWriter2 fileWriter2_0 = new FileWriter2("*rx~_;H'6SN$ilNZuk");
      FileWriter2 fileWriter2_1 = new FileWriter2();
      fileWriter2_1.writeLog("*rx~_;H'6SN$ilNZuk");
      fileWriter2_1.writeJson("0a|p?x");
      fileWriter2_0.writeLog(636, "j{");
      assertNotSame(fileWriter2_0, fileWriter2_1);
  }
}
